src/Drivers/nlpPriDec_ex9_sparse_raja.cpp

```diff
@@ -0,0 +1,222 @@
+#include "nlpPriDec_ex9_sparse_raja.hpp"
+#include "hiopNlpFormulation.hpp"
+#include "hiopAlgFilterIPM.hpp"
+
+#include <RAJA/RAJA.hpp>
+/*
+  todo:
+  - generate S vectors \xi (each of size nS) from U[-0.25, 0.25]
+*/
+
+using ex9_raja_exec = hiop::hiop_raja_exec;
+using ex9_raja_reduce = hiop::hiop_raja_reduce;
+using namespace hiop;
+
+	
+PriDecMasterProblemEx9Sparse::
+PriDecMasterProblemEx9Sparse(size_t nx, size_t ny, size_t nS, size_t S, std::string mem_space) 
+  : nx_(nx), 
+    ny_(ny),
+    nS_(nS),
+    S_(S),
+    mem_space_(mem_space)
+{
+  assert(nx==ny);
+  y_ = new double[ny_];
+  sol_ = new double[nx_];
+  obj_ = 1e20;
+  basecase_ = new PriDecBasecaseProblemEx9(nx_);
+  nc_ = nx_;
+};
+
+PriDecMasterProblemEx9Sparse::~PriDecMasterProblemEx9Sparse()
+{
+  delete[] y_;
+  delete[] sol_; 
+  delete basecase_;
+};
+	
+hiop::hiopSolveStatus
+PriDecMasterProblemEx9Sparse::solve_master(hiopVector& x,
+                                           const bool& include_r,
+                                           const double& rval/*=0*/, 
+                                           const double* grad/*=0*/,
+                                           const double*hess /*=0*/,
+                                           const char* master_options_file/*=nullptr*/)
+{
+  obj_=-1e+20;
+  hiopSolveStatus status;
+  if(basecase_==NULL) {
+    basecase_ = new PriDecBasecaseProblemEx9(nx_);
+  }
+  basecase_->set_include(include_r);
+
+  if(include_r) {
+    assert(basecase_->quad_is_defined());
+  }
+  
+  hiopNlpSparse nlp(*basecase_, master_options_file);
+
+  //
+  // any of the options below can be overwritten by specifying them in the 'hiop_pridec_master.options' file
+  //
+  //nlp.options->SetStringValue("compute_mode", "hybrid");
+  //nlp.options->SetStringValue("dualsUpdateType", "linear");
+
+  //nlp.options->SetStringValue("fixed_var", "relax");
+  nlp.options->SetStringValue("Hessian", "analytical_exact");
+  nlp.options->SetStringValue("KKTLinsys", "xdycyd");
+  nlp.options->SetStringValue("compute_mode", "cpu");
+  //nlp.options->SetStringValue("compute_mode", "hybrid");
+  //nlp.options->SetStringValue("mem_space", mem_space.c_str());
+
+  nlp.options->SetIntegerValue("verbosity_level", 1);
+  nlp.options->SetNumericValue("mu0", 1e-1);
+  //nlp.options->SetNumericValue("tolerance", 1e-5);
+  
+  hiopAlgFilterIPMNewton solver(&nlp);
+
+  status = solver.run();
+
+  obj_ = solver.getObjective();
+  double* x_vec = x.local_data();
+  solver.getSolution(x_vec);
+
+  if(status<0) {
+    printf("solver returned negative solve status: %d (with objective is %18.12e)\n", status, solver.getObjective());
```

Comment:
all the outputting inside the solver needs to be done via hiopLogger... except the main driver where printf/cout/cerr is ok.

Reply:
Using nlp log to print instead.